### PR TITLE
Added support for numbers with mantissa

### LINKED ISSUE #883: I want "round" function might support numbers like "1.056E7b" or Vividus can correctly work with "big" numbers like 10 560 000 and alike without adding 'b'
There is a problem with numbers which have more than 5 digits (and which might look like "1.056E7b" after vividus` interpreting)
At the moment we use such with adding 'b' after it. 

Example: 
```
${x} * ${y} = ${x*y} in order to use variable ${x*y} in further calculations we add 'b' => '${x*y}b' 
132000*80 = 10 560 000 == 1.056E7
```

It would be great if mentioned numbers type will correctly collaborate with "round" functionality. 

Also I would like to notice that number in the variable '${x*y}b' (10 560 000 == 1.056E7) is sent with wrong value into field in used UI. 
Example: 
1) save expected value in '${x*y}'
2) using step 'When I enter `${x*y}b` in field located `By.id(bv1)' 
3) send request with value
it does not sent 1.056E7 but delete everything after letter (looks like 1.056) and automattically round number => 1.06
but in the same time, value of this variable is used in the possible next steps with its value = 1.056E7

### PR DESCRIPTION
**Pull request by BViktoryia**
Closes #883

### CODE REVIEW

**Review by ikalinin1 — APPROVED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java [hunk 1]
```diff
@@ -38,8 +38,8 @@
     {
         String roundingModes = Stream.of(RoundingMode.values()).map(e -> e.name().toLowerCase())
                 .collect(Collectors.joining("|"));
-        String pattern = String.format("^round(?:\\((-?\\d+(?:\\.\\d*)?)(?:,\\s*(\\d+))?(?:,\\s*(%s))?\\))$",
-            roundingModes);
+        String pattern = String.format("^round(?:\\((-?\\d+(?:\\.\\d*)?[Ee]?-?\\+?\\d*)"
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java:41]: this regex will also match the following invalid strings:
 - `0.15237E`
 - `0.15237E-+2`
 - `0.15237-2`
 - `0.15237+2`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java (hunk 1 above)
Comment (BViktoryia) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java]: fixed

Comment (valfirst) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java]: I believe all items from the comment above should be added to negative unit tests

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java [hunk 2]
```diff
@@ -38,8 +38,8 @@
     {
         String roundingModes = Stream.of(RoundingMode.values()).map(e -> e.name().toLowerCase())
                 .collect(Collectors.joining("|"));
-        String pattern = String.format("^round(?:\\((-?\\d+(?:\\.\\d*)?)(?:,\\s*(\\d+))?(?:,\\s*(%s))?\\))$",
-            roundingModes);
+        String pattern = String.format("^round(?:\\((-?\\d+(?:\\.\\d*)?(?:[Ee]+-?\\d+)?)(?:,\\s*(\\d+))?"
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java:41]: the pattern won't match `0.15237E+2` which is a valid number

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java (hunk 1 above)
Comment (BViktoryia) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java]: added

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java (hunk 2 above)
Comment (BViktoryia) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java]: fixed

**Review by valfirst — APPROVED** [review]

**Pull request merged**